ethpm_cli/scraper.py

```diff
@@ -73,12 +112,16 @@ def initialize_ethpm_dir(ethpm_dir: Path, w3: Web3) -> None:
 
 
 def update_chain_data(
-    chain_data_path: Path, block_number: int, manifests: Dict[Address, Dict[str, str]]
+    chain_data_path: Path,
+    from_block: int,
+    to_block: int,
+    manifests: Dict[Address, Dict[str, str]],
 ) -> None:
     chain_data = json.loads(chain_data_path.read_text())
 
-    all_scraped_blocks = get_scraped_blocks(chain_data_path)
-    updated_blocks = blocks_to_ranges(set(all_scraped_blocks + [block_number]))
+    old_scraped_blocks = get_scraped_blocks(chain_data_path)
+    new_scraped_blocks = list(range(from_block, to_block))
+    updated_blocks = blocks_to_ranges(set(old_scraped_blocks + new_scraped_blocks))
 
     chain_data_updated_blocks = assoc(chain_data, "scraped_blocks", updated_blocks)
     chain_data_path.write_text(f"{json.dumps(chain_data_updated_blocks, indent=4)}\n")
```

Comment:
you might consider using the tempdir/tempfile approach for this as well to ensure that you don't end up with a corrupt file if the process dies half-way through writing.

- copy file
- update the copy
- replace original file with updated copy